connectors/src/api/webhooks/webhook_github.ts

```diff
@@ -250,6 +254,14 @@ const _webhookGithubAPIHandler = async (
             jsonBody.pull_request.number,
             res
           );
+        } else if (jsonBody.action === "closed") {
```

Comment:
```suggestion
        } else if (jsonBody.action === "closed" && jsonBody.pull_request.merged) {
```

Reply:
avoid syncing on closed without merging PRs

Reply:
If you apply this suggestion I think assertNever will break. Needs to be 2 ifs I think ?